Sources/SwiftDocC/Infrastructure/Link Resolution/PathHierarchy+PathComponent.swift

```diff
@@ -170,56 +174,64 @@ private struct PathComponentScanner {
     }
     
     mutating func scanPathComponent() -> Substring {
-        // If the next component is an operator, parse the full operator before splitting on "/" ("/" may appear in the operator name)
-        if remaining.unicodeScalars.prefix(3).allSatisfy(\.isValidOperatorHead) {
-            guard let operatorEndIndex = remaining.firstIndex(of: Self.operatorEnd) else {
-                defer { remaining.removeAll() }
-                return remaining
-            }
-            
-            var component = remaining[..<operatorEndIndex]
-            remaining = remaining[operatorEndIndex...]
-            
-            guard let index = remaining.firstIndex(of: Self.separator) else {
-                defer { remaining.removeAll() }
-                return component + remaining
-            }
-            
-            component += remaining[..<index]
-            remaining = remaining[index...].dropFirst() // drop the slash
-            return component
-            
+        // If the next component is a Swift operator, parse the full operator before splitting on "/" ("/" may appear in the operator name)
+        if remaining.unicodeScalars.prefix(3).allSatisfy(\.isValidSwiftOperatorHead) {
+            return scanUntil(index: remaining.firstIndex(of: Self.swiftOperatorEnd))
+                 + scanUntilSeparatorAndThenSkipIt()
+        }
+        
+        // If the next component is a C++ operator, parse the full operator before splitting on "/" ("/" may appear in the operator name)
+        if remaining.starts(with: Self.cxxOperatorPrefix),
+           remaining.unicodeScalars.dropFirst(Self.cxxOperatorPrefixLength).first?.isValidCxxOperatorSymbol == true
+        {
+            return scan(length: Self.cxxOperatorPrefixLength)
+                 + scanUntil(index: remaining.unicodeScalars.firstIndex(where: { !$0.isValidCxxOperatorSymbol }))
+                 + scanUntilSeparatorAndThenSkipIt()
         }
         
         if remaining.first == Self.separator {
-            guard let index = remaining.firstIndex(where: { $0 != Self.separator }) else {
-                defer { remaining.removeAll() }
-                return remaining
-            }
-            let slashPrefix = remaining[..<index]
-            remaining = remaining[index...]
-            return slashPrefix + scanPathComponent()
+            return scanUntil(index: remaining.firstIndex(where: { $0 != Self.separator }))
```

Comment:
This isn't a new change but out of curiosity in what situations would we be parsing something that looks that starts with a `/` and want to preserve that?

Reply:
There are at least two Swift operators that start with a leading slash (`/(_:_:)` and `/=(_:_:)`). 

If the developer writes a link like `/(_:_:)` we want to parse that link as a single "/(\_:\_:)" component, and if they wrote `Something//(_:_:)` we want to parse that link as a "Something" component and a "/(\_:\_:)" component.

Reply:
Is that not captured by `if remaining.unicodeScalars.prefix(3).allSatisfy(\.isValidSwiftOperatorHead) {`

Reply:
Ah! Yes, you are correct. 

Looking at this in context, the reason for preserving the slash here is an implementation detail private to this file for how to identify an "absolute link" (starting with a slash). Since this code in the path component parser doesn't know/track if its at the start of the original string or not, it includes the leading slash in certain components and defer to [this code in the path parser](https://github.com/swiftlang/swift-docc/blob/main/Sources/SwiftDocC/Infrastructure/Link%20Resolution/PathHierarchy%2BPathComponent.swift#L117-L126) to identify if the link is absolute and trim the leading slash. 

Together, this means that `"/ModuleName/Something"` gets split into 
```
(componentSubstrings: ["ModuleName", "Something"], isAbsolute: true)
```
without the leading slash in the first component's name, whereas `"/(_:_:)"` gets split into
```
(componentSubstrings: ["/(_:_:)"], isAbsolute: false)
```
with the leading slash in the first component.

The path hierarchy then uses the `isAbsolute` value to determine if it should start the search at the top or relative to the current page.

Reply:
OK It's not obvious from the local context that it's important to preserve leading separators, so it would be good to add a comment indicating that this is the reason.

Reply:
I added more documentation—with examples—and inline code comments in https://github.com/swiftlang/swift-docc/pull/1003/commits/fe4f64137f90243f8ee22b468e336899461ca166 to describe how this works.